App/Controllers/MatchesController.php

```diff
@@ -18,17 +18,28 @@ public function create()
      */
     public function store(ServerRequest $request)
     {
+        $matches = new SummonersModel();
         $data = $request->getParsedBody();
 
-        $leagueOfLegendsService = new LeagueOfLegendsService($data['name']);
+        $leagueOfLegendsService = new LeagueOfLegendsService();
 
-        $summonerData = $leagueOfLegendsService->getSummonerData();
+        $summonerData = $leagueOfLegendsService->getSummonerData($data['name']);
 
-        $identity = $summonerData['summoner_id'];
+        $matchList = $leagueOfLegendsService->matchlist($summonerData['summoner_id']);
+
+        //* @todo might need to unset('region', 'platformId', 'champion', 'queue', 'season', 'timestamp', 'role')
+        for ($matchIndex = $matchList->raw()['matches']['startIndex'];
+             $matchIndex <= $matchList->raw()['matches']['endIndex'];
+             $matchIndex++) {
+
+            $result = $matchList->raw()['matches'][$matchIndex];
+
+            $result['summoner_id'] = $summonerData['summoner_id'];
+
+            $matches->fill($result);
+        }
```

Comment:
The logic to fill in the matches table can be moved into the summoner controller for now. Nothing wrong with having the `SummonersModel()` class and the `$leagueOfLegendsService` `$matchlist` variable in there as well. If we hit a point where both controllers are doing similar work we can break that logic out into an additional class so it can be shared between the two.